### PR TITLE
Staging/user profile

### PR DESCRIPTION
**Pull request by Enjoy2Live**
user-profile first release

### CODE REVIEW

File: apps/app/next.config.js [hunk 1]
```diff
@@ -3,9 +3,11 @@ const withPlugins = require('next-compose-plugins');
 const imagesPlugin = require('next-optimized-images');
 const withTM = require('next-transpile-modules')([
   '@devlaunchers/ideaspace',
+  '@devlaunchers/user-profile',
   '@devlaunchers/site-projects',
   '@devlaunchers/dev-recruiters',
   '@devlaunchers/website',
+  '@devlaunchers/user-profile',
```
Comment (Enjoy2Live) [inline apps/app/next.config.js:10]: @judeamos I see two @devlaunchers/user-profile, why?

File: apps/user-profile/.eslintrc-example.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    // any conficts between eslint / prettier
+    "@devlaunchers/eslint-config-bases/prettier",
+  ],
+  rules: {
+    // Specific global rules for your app or package
+  },
+  overrides: [
+    // Specific file rules for your app or package
+  ],
+};
```
Comment (Enjoy2Live) [inline apps/user-profile/.eslintrc-example.js:38]: let's change this file from ".eslintrc-example.js" to ".eslintrc.js"

File: apps/user-profile/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
+}
```
Comment (Enjoy2Live) [inline apps/user-profile/.eslintrc.json:3]: remove this file

File: apps/user-profile/.flux.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+version: 1
+patchUpdated:
+  generators:
+    - command: kustomize build .
+  patchFile: flux-patch.yaml
```
Comment (Enjoy2Live) [inline apps/user-profile/.flux.yaml:5]: I don't think we need this file right @chungthuang ?

File: apps/user-profile/.gitignore copy [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (Enjoy2Live) [inline apps/user-profile/.gitignore copy:28]: why two .gitignore files?

File: apps/user-profile/.husky/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+_
```
Comment (Enjoy2Live) [inline apps/user-profile/.husky/.gitignore:1]: remove `.husky` folder

File: apps/user-profile/.travis.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    target_branch: master
+    on:
+      branch: release
+  # deploy Github page to gh-pages branch
+  - provider: pages
+    skip_cleanup: true
+    github_token: $GITHUB_TOKEN
+    local_dir: build
+    on:
+      branch: master
```
Comment (Enjoy2Live) [inline apps/user-profile/.travis.yml:26]: we don't need this right @chungthuang ?

File: apps/user-profile/README.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
+- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+
+You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
+
+## Deploy on Vercel
+
+The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+
```
Comment (Enjoy2Live) [inline apps/user-profile/README.md:33]: this seems general let's remove it or make it more suitable for this app

File: apps/user-profile/README copy.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+We are using semantic versioning to tag release. Follow https://github.com/semantic-release/semantic-release#commit-message-format
+to format the commit messages.
+
+Once you are ready to create a new release, create a PR to merge main branch to release branch.
+
+.
+
+---
+
+## LicenseS
```
Comment (Enjoy2Live) [inline apps/user-profile/README copy.md:59]: carbon copy of the main readme file, let's make it more suitable to this workspace

File: apps/user-profile/next.config copy.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    // your project has ESLint errors.
+    // we have too many errors if you run npm run lint ,but after bug fixes we could enforce this.
+    ignoreDuringBuilds: true,
+  },
+  output: 'standalone',
+  experimental: {
+    externalDir: true,
+  },
+};
+module.exports = withPlugins([[imagesPlugin], [withTM]], nextConfig);
```
Comment (Enjoy2Live) [inline apps/user-profile/next.config copy.js:45]: let's remove this

File: apps/user-profile/package.json [hunk 11]
```diff
@@ -82,7 +86,9 @@
     "styled-components": "^5"
   },
   "release": {
-    "branches": ["release"],
+    "branches": [
+      "release"
+    ],
     "plugins": [
       "@semantic-release/commit-analyzer",
       "@semantic-release/release-notes-generator",
```
Comment (Enjoy2Live) [inline apps/user-profile/package.json:94]: not needed

File: apps/user-profile/package.json [hunk 12]
```diff
@@ -73,7 +72,12 @@
...
-  "browserslist": [">0.2%", "not dead", "not ie <= 11", "not op_mini all"],
+  "browserslist": [
+    ">0.2%",
+    "not dead",
+    "not ie <= 11",
+    "not op_mini all"
+  ],
   "lint-staged": {
     "*.js": "eslint '*/**/*.{js,jsx}' --fix",
     "*.{js,css,md}": "prettier --write"
```
Comment (Enjoy2Live) [inline apps/user-profile/package.json:83]: not needed

File: apps/user-profile/src/components/common/Button/Button.js
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Button/Button.js:1]: let's delete this file and use the shared Button component only

File: apps/user-profile/src/components/common/Button/index.js
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Button/index.js:1]: let's delete this file and use the shared Button component only

File: apps/user-profile/src/components/common/Forms/Checkbox/CheckboxField.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+import { CheckboxContainer } from './StyledCheckboxField';
+const InputField = ({ customLabel, label, onChange, required, checked, children }) => {
+    return (
+        <CheckboxContainer>
+            <Checkbox
+                label={customLabel}
+                onChange={onChange}
+                required={required}
+                checked={checked}
+            />
```
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Forms/Checkbox/CheckboxField.js:11]: let's delete this and use the shared Checkbox component only

File: apps/user-profile/src/components/common/Modal/Modal.js
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Modal/Modal.js:1]: Delete this and use the Dialog shared component.

File: apps/user-profile/src/components/common/Modal/StyledModal.js
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Modal/StyledModal.js:1]: Delete this and use the Dialog shared component.

File: apps/user-profile/src/components/common/Modal/index.js
Comment (Enjoy2Live) [inline apps/user-profile/src/components/common/Modal/index.js:1]: Delete this and use the Dialog shared component.

File: apps/user-profile/src/components/modules/UserOnboardingModal/DiscordOnboarding/DiscordOnboarding.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+
+
+
+
+
+
+export default function DiscordOnboarding() {
+    return(<></>);
+}
```
Comment (Enjoy2Live) [inline apps/user-profile/src/components/modules/UserOnboardingModal/DiscordOnboarding/DiscordOnboarding.js:14]: why empty component?

File: apps/user-profile/src/pages/_app.js [hunk 15]
```diff
@@ -0,0 +1,83 @@
+import Head from 'next/head';
+import { Router, useRouter } from 'next/router';
+import React from 'react';
+import { ToastContainer } from 'react-toastify';
+import { ThemeProvider } from 'styled-components';
+
+import GlobalStyle from '../styles/globals';
```
Comment (Enjoy2Live) [inline apps/user-profile/src/pages/_app.js:7]: GlobalStyle should be imported from "@devlaunchers/components/src/styles/styles"

File: apps/user-profile/src/utils/EnvironmentVariables.js
Comment (Enjoy2Live) [inline apps/user-profile/src/utils/EnvironmentVariables.js:1]: no one uses this file anymore, please delete

File: apps/user-profile/tsconfig.json [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  "exclude": ["**/node_modules", "**/.*/", ".next"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    "**/*.js",
+    "**/*.jsx",
+    "**/*.mjs",
+    "**/*.json"
+, ".eslintrc-example.js"  ]
```
Comment (Enjoy2Live) [inline apps/user-profile/tsconfig.json:33]: change this to ".eslintrc.js"

File: apps/user-profile/wrangler.toml
Comment (Enjoy2Live) [inline apps/user-profile/wrangler.toml:1]: we don't need this right @chungthuang

File: packages/UI/package.json [hunk 17]
```diff
@@ -13,7 +13,7 @@
     "clsx": "2.0.0",
     "constate": "^3.3.2",
     "formik": "^2.2.9",
-    "framer-motion": "6.5.1",
+    "framer-motion": "6.2.4",
```
Comment (Enjoy2Live) [inline packages/UI/package.json:16]: why downgrading?

File: packages/UI/src/context/UserDataContext.js [hunk 18]
```diff
@@ -3,20 +3,54 @@ import { useCallback, useEffect, useState } from 'react';
...
-  email: '',
-  bio: '',
-  profilePictureUrl: '',
+  id: 6,
+  name: 'Kris Gano',
+  username: 'Kris_Gano',
+  email: 'krisgano@devlaunchers.com',
+  bio: 'A kris bio',
+  profilePictureUrl:
+    'https://lh3.googleusercontent.com/a/AATXAJxZ3nb3nA60iLAv1-RqQizgB871aHjJp4LTwi55=s96-c',
```
Comment (Enjoy2Live) [inline packages/UI/src/context/UserDataContext.js:12]: why defaulting to these values?

File: packages/UI/src/context/UserDataContext.js [hunk 19]
```diff
@@ -3,20 +3,54 @@ import { useCallback, useEffect, useState } from 'react';
...
+    },
+    {
+      id: 4,
+      published_at: '2021-08-11T05:36:30.868Z',
+      created_at: '2021-08-11T05:36:25.579Z',
+      updated_at: '2021-08-11T05:36:31.097Z',
+      interest: 'UI/UX',
+      categories: [],
+    },
+  ],
```
Comment (Enjoy2Live) [inline packages/UI/src/context/UserDataContext.js:53]: why defaulting to these values?

**Review by Enjoy2Live — COMMENTED** [review]
we need to make sure files that are not imported anywhere should be deleted, there's a lot of files here that are copied from the original app but really don't serve any purpose

**Comment by Enjoy2Live** [comment]
@judeamos when will this make it to master?

**Review by judeamos — APPROVED** [review]
All looks good to me

**Pull request merged**